### ISSUE TITLE
Directory `.next/static/css` is no longer being generated

### ISSUE THREAD

**Issue by royjosefsson** [vanilla-extract-css/vanilla-extract#1491]
## Issue Summary
Upgrading `@vanilla-extract/next-plugin` from version `2.3.7` to `2.4.6`, 
results in the `.next/static/css` directory no longer being generated.

## Steps to Reproduce
1. Run `npm run dev` _(using Next.js page-router)_

## Expected Behavior
Running `npm run dev` in version `2.4.6` generates `.next/static/css` along with its files inside

## Actual Behavior
Running `npm run dev` in version `2.4.6` doesn't generate `.next/static/css`

## Severity
- [X] Blocker
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### Reproduction

https://github.com/royjosefsson/vanilla-extract-issue/tree/main

### System Info

```shell
System:
  OS: macOS 15.0
  CPU: (10) arm64 Apple M1 Max
  Memory: 2.92 GB / 32.00 GB
  Shell: 5.9 - /bin/zsh
Binaries:
  Node: 20.12.2 - ~/.n/bin/node
  npm: 10.5.0 - ~/.n/bin/npm
  bun: 1.1.24 - ~/.bun/bin/bun
Browsers:
  Chrome: 129.0.6668.101
  Safari: 18.0
npmPackages:
  @vanilla-extract/css: 1.14.2 => 1.14.2 
  @vanilla-extract/dynamic: 2.1.0 => 2.1.0 
  @vanilla-extract/jest-transform: 1.1.1 => 1.1.1 
  @vanilla-extract/next-plugin: 2.3.7 => 2.3.7
```

### Used Package Manager

npm

### Validations

- [X] Check that there isn't [already an issue](https://github.com/vanilla-extract-css/vanilla-extract/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by royjosefsson** [comment]
@mrm007 
Would you mind looking at this, is this a bug or are we doing this intentionally?

**Comment by mrm007** [comment]
AFAIK the Next.js integration has not been tested with Next.js 14 and above. I’m surprise the plugin version 2.3.7 still works.

@askoufis might know more.

**Comment by mrm007** [comment]
There's also this issue with Turbopack https://x.com/0_001x_dev/status/1848744288743522561

**Comment by askoufis** [comment]
@royjosefsson This is intentional behaviour that was implemented in v2.4.0 of the next plugin (https://github.com/vanilla-extract-css/vanilla-extract/pull/1365).

**Comment by royjosefsson** [comment]
@askoufis 

**Goal:**  
Enable loading theme-specific stylesheets dynamically using a `<link>` tag in the `<head>`.

---

### **v2.3.0**  
- **Development and Production:**  
  CSS files were generated inside `.next/static/css`, with each file being split into smaller files.👏

---

### **v2.3.7**  
- **Development:**  
  CSS files were generated inside `.next/static/css`, with each file being split.👏
- **Production:**  
  CSS was generated inside `.next/static/css`, but the files were not split.😞

---

### **v2.4.x**  
- **Development:**  
  Where is the CSS being generated?🤔 
- **Production:**  
  Where is the CSS being generated?🤔

**Comment by askoufis** [comment]
@royjosefsson 

In 2.4.x, CSS is now injected into the DOM, for reasons described in #1365. In production builds (I tested it in your reproduction), CSS is emitted to `.next/static/css`.

**Comment by askoufis** [comment]
@royjosefsson Are you still experiencing this issue? As far as I can tell the `.next/static/css` _is_ being generated.

**Comment by royjosefsson** [comment]
@askoufis
Unfortunately, yes. 
How is the .next/static/css being generated on your end? 
Could you share your step-by-step process?

Here is a screendump of all the files that's being generated for me:

![Image](https://github.com/user-attachments/assets/189fe975-32fe-4251-a321-2748bfc4e4e1)

**Comment by royjosefsson** [comment]
@askoufis
From what I understand, this behavior is intentional.

CSS is no longer being generated in `.next/static/css` when running `npx next dev`; it’s only generated in `.next/static/css` during `npx next build`.

We could consider closing this issue and I'll fix my code to this breaking change

**Comment by askoufis** [comment]
> @askoufis
> From what I understand, this behavior is intentional.
> 
> CSS is no longer being generated in `.next/static/css` when running `npx next dev`; it’s only generated in `.next/static/css` during `npx next build`.
> 
> We could consider closing this issue and I'll fix my code to this breaking change

Yes, this is what I have been saying above. This was an intentional change in behaviour. Is there a reason you need the static/CSS files during development?
